src/utils/common.ts

```diff
@@ -1,26 +1,32 @@
+import { isRouteErrorResponse } from '@remix-run/react';
+import { isEcomSDKError } from '~/api/types';
+
 export function getUrlOriginWithPath(url: string) {
     const { origin, pathname } = new URL(url);
     return new URL(pathname, origin).toString();
 }
 
-export function toError(value: unknown): Error {
-    if (value instanceof Error) {
-        return value;
+/**
+ * Try to find some error message in unknown value.
+ *
+ * Handles remix ErrorResponse and wix ecom platform EcomSDKError error types.
+ */
+export function getErrorMessage(value: unknown): string | undefined {
+    if (isRouteErrorResponse(value)) {
+        return String(value.data);
```

Comment:
Looking at the implementation of `ErrorResponse` I'm not sure this is the correct way to handle this.
https://github.com/remix-run/react-router/blob/react-router-dom%406.14.1/packages/router/utils.ts#L1472-L1495

If the object has an error property, we should use value.error.message, otherwise we should use `String(value.data)`.

Reply:
you used wrong version in a link.
according to types, there is no `error` property there:
https://github.com/remix-run/react-router/blob/%40remix-run/router%401.19.2/packages/router/utils.ts#L1656-L1693

however, added additional check if `value.data` is instance of `Error`

Reply:
Ok, so error is private, but why do you need to check if value.data is an instance of Error? They're converting it to a string in the version that you linked.

Reply:
my bad, fixed